lms/templates/student_account/register.underscore

```diff
@@ -29,11 +29,17 @@
             <%
             _.each( context.providers, function( provider) {
                 if ( provider.registerUrl ) { %>
-                    <button type="button" class="button button-primary button-<%- provider.id %> login-provider register-<%- provider.id %>" data-provider-url="<%- provider.registerUrl %>">
-                        <span class="icon fa <%- provider.iconClass %>" aria-hidden="true"></span>
-                        <span aria-hidden="true"><%- provider.name %></span>
-                        <span class="sr"><%- _.sprintf( gettext("Create account using %(providerName)s."), {providerName: provider.name} ) %></span>
-                    </button>
+                    <div class="button-container" <% if ( provider.buttonColor ) { %>style="color: <%- provider.buttonColor %>; background-color: <%- provider.buttonColor %>"<% } %>>
```

Comment:
I still think a provider specific class should be added here and the styling handled by updating a third party auth Sass partial instead of using inline styles. It will make the code much more flexible and enable full editing of button states, and it is best practice to avoid inline styles unless absolutely necessary.